spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -407,6 +428,216 @@ public void close() throws IOException {
     }
   }
 
+  private static class UnpartitionedDeltaWriter extends BaseDeltaWriter {
```

Comment:
Both `UnpartitionedDeltaWriter` and `PartitionedDeltaWriter` share the `delete` implementation but `insert` and `update` methods are different. We could refactor these two into a single classs but that would raise questions about extra method calls. We could go with an accessor and hope it will be inlined by JIT but we can't know for sure.